crates/evm/src/fuzz/mod.rs

```diff
@@ -216,6 +213,61 @@ impl<'a> FuzzedExecutor<'a> {
 
         result
     }
+
+    pub fn single_fuzz(
+        &self,
+        state: &EvmFuzzState,
+        address: Address,
+        should_fail: bool,
+        calldata: ethers::types::Bytes,
+    ) -> Result<FuzzOutcome, TestCaseError> {
+        let call = self
+            .executor
+            .call_raw(self.sender, address, calldata.0.clone(), 0.into())
+            .map_err(|_| TestCaseError::fail(FuzzError::FailedContractCall))?;
+        let state_changeset = call
+            .state_changeset
+            .as_ref()
+            .ok_or_else(|| TestCaseError::fail(FuzzError::EmptyChangeset))?;
+
+        // Build fuzzer state
+        collect_state_from_call(
+            &call.logs,
+            state_changeset,
+            state.clone(),
+            &self.config.dictionary,
+        );
+
+        // When assume cheat code is triggered return a special string "FOUNDRY::ASSUME"
+        if call.result.as_ref() == ASSUME_MAGIC_RETURN_CODE {
+            return Err(TestCaseError::reject(FuzzError::AssumeReject))
+        }
+
+        let breakpoints =
+            call.cheatcodes.clone().map_or(Default::default(), |cheats| cheats.breakpoints);
+
+        let success =
+            self.executor.is_success(address, call.reverted, state_changeset.clone(), should_fail);
+
+        if success {
+            Ok(FuzzOutcome::Case(CaseOutcome {
+                case: FuzzCase { calldata, gas: call.gas_used, stipend: call.stipend },
+                gas_used: call.gas_used,
+                stipend: call.stipend,
+                traces: call.traces,
+                coverage: call.coverage,
+                debug: call.debug,
+                breakpoints,
+            }))
+        } else {
+            Ok(FuzzOutcome::CounterExample(CounterExampleOutcome {
+                debug: call.debug.clone(),
+                exit_reason: call.exit_reason,
+                counterexample: (calldata, call),
+                breakpoints,
```

Comment:
interesting, didn't we try and shrink here before?

Reply:
Nice catch, but not sure it's shrinking in this case, looking at the proptest docs it seems to only shrink in the case of a `TestError` https://proptest-rs.github.io/proptest/proptest/tutorial/test-runner.html
They are bubbled up only on https://github.com/foundry-rs/foundry/pull/5692/files#diff-ba41d5025a0a94dcc6ac6d03c5d3a934f11a9df435e9b771710ea67c9c812c00R163-R165 but please correct me if i'm wrong